telegram/ext/_baseupdateprocessor.py

```diff
@@ -0,0 +1,151 @@
+        self._max_concurrent_updates = max_concurrent_updates
+        if self.max_concurrent_updates < 1:
+            raise ValueError("`max_concurrent_updates` must be a positive integer!")
+        self._semaphore = BoundedSemaphore(self.max_concurrent_updates)
+
+    @property
+    def max_concurrent_updates(self) -> int:
+        """:obj:`int`: The maximum number of updates that can be processed concurrently."""
+        return self._max_concurrent_updates
+
+    @abstractmethod
+    async def do_process_update(
+        self,
+        update: object,
+        coroutine: "Awaitable[Any]",
+    ) -> None:
+        """Custom implementation of how to process an update. Must be implemented by a subclass.
+
+        Warning:
+            This method will be called by :meth:`process_update`. It should *not* be called
+            manually.
+
+        Args:
+            update (:obj:`object`): The update to be processed.
+            coroutine (:obj:`Awaitable`): The coroutine that will be awaited to process the update.
+        """
+
+    @abstractmethod
+    async def initialize(self) -> None:
+        """Initializes the processor so resources can be allocated. Must be implemented by a
+        subclass.
+
+        .. seealso::
+            :meth:`shutdown`
+        """
+
+    @abstractmethod
+    async def shutdown(self) -> None:
+        """Shutdown the processor so resources can be freed. Must be implemented by a subclass.
+
+        .. seealso::
+            :meth:`initialize`
+        """
+
+    async def process_update(
+        self,
+        update: object,
+        coroutine: "Awaitable[Any]",
+    ) -> None:
+        """Calls :meth:`do_process_update` with a semaphore to limit the number of concurrent
+        updates.
+
+        Args:
+            update (:obj:`object`): The update to be processed.
+            coroutine (:obj:`Awaitable`): The coroutine that will be awaited to process the update.
+        """
+        async with self._semaphore:
+            await self.do_process_update(update, coroutine)
+
+    async def __aenter__(self) -> "BaseUpdateProcessor":
+        """Simple context manager which initializes the Processor."""
+        try:
+            await self.initialize()
+            return self
+        except Exception as exc:
+            await self.shutdown()
+            raise exc
+
+    async def __aexit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
+        exc_val: Optional[BaseException],
+        exc_tb: Optional[TracebackType],
+    ) -> None:
+        """Shutdown the Processor from the context manager."""
+        await self.shutdown()
+
+
+class SimpleUpdateProcessor(BaseUpdateProcessor):
+    """Instance of :class:`telegram.ext.BaseUpdateProcessor` that immediately awaits the
+    coroutine, i.e. does not apply any additional processing. This is used by default when
+    :attr:`telegram.ext.ApplicationBuilder.concurrent_updates` is :obj:`int`.
+
```

Comment:
Don't we want to add `Args` to this docstring as well? Or maybe change the .rst file to `:inherit-members:` or something like that

Reply:
Ok nvm, that didn't fix it, maybe just write add a docstring here saying to refer `BaseUpdateProcessor` for args to pass? @clot27

Reply:
We don't list arguments of the base classes anywhere else either, do we? The base classes are listed in any case, which suffices IMO

Reply:
alright then